src/compiler/checker.ts

```diff
@@ -6867,20 +6867,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                 }
                 else {
-                    const isInstantiationExpressionType = !!(getObjectFlags(type) & ObjectFlags.InstantiationExpressionType);
-                    if (isInstantiationExpressionType) {
-                        const instantiationExpressionType = type as InstantiationExpressionType;
-                        if (isTypeQueryNode(instantiationExpressionType.node)) {
-                            const typeNode = serializeExistingTypeNode(context, instantiationExpressionType.node);
-                            if (typeNode) {
-                                return typeNode;
-                            }
-                        }
-                        if (context.visitedTypes?.has(typeId)) {
-                            return createElidedInformationPlaceholder(context);
-                        }
-                        return visitAndTransformType(type, createTypeNodeFromObjectType);
-                    }

```

Comment:
```ts
                    const isInstantiationExpressionType = !!(getObjectFlags(type) & ObjectFlags.InstantiationExpressionType);
                    if (isInstantiationExpressionType) {
                        const instantiationExpressionType = type as InstantiationExpressionType;
                        const existing = getEffectiveTypeAnnotationNode(instantiationExpressionType.node)!;
                        if (isTypeQueryNode(instantiationExpressionType.node) && getTypeFromTypeNode(existing) === type && existingTypeNodeIsNotReferenceOrIsReferenceWithCompatibleTypeArgumentCount(existing, type)) {
                            const typeNode = serializeExistingTypeNode(context, instantiationExpressionType.node);
                            if (typeNode) {
                                return typeNode;
                            }
                        }
                        if (context.visitedTypes?.has(typeId)) {
                            return createElidedInformationPlaceholder(context);
                        }
                        return visitAndTransformType(type, createTypeNodeFromObjectType);
                    }
```
Should allow still reusing the `typeof symbolName<Thing>` input node when that node isn't affected by further instantiation. Like in the `typeof Array<number>` case.

Reply:
Cool, I'll try it!

Reply:
Well, this will have to be put up into the other branch now that they have symbols but I think it should work.

Reply:
Seems like `getEffectiveTypeAnnotationNode` is always returning `undefined`; is there a bit I'm missing?

Reply:
Yeah, the edit I made a few seconds after posting swapping it to just be `getTypeFromTypeNode` (which may need to be conditioned on the `isTypeQueryNode` check). 😜

Reply:
Thanks, that did make the baselines no longer change for any test in this PR. Is `getTypeFromTypeNode(existing) === type` enough, though? I'm thinking of a case where the type comes from another file's instantiation expression but the node shouldn't be reused in another file.

But, removing the check (to see what it does) does show that it's doing something.